From fd9eeda3e914002091d5a8442abbe28ea2878812 Mon Sep 17 00:00:00 2001
From: nxf36763 <ashish.manohar@nxp.com>
Date: Fri, 21 Mar 2025 15:12:58 +0530
Subject: [PATCH 3/4] [artf1201762]: Fix for GMS compliancy issue

- Fix for CTS issue
- powerstate update for tech & proto route
- PN7220 chipset power state is updated to support in all screen state.

Change-Id: I41234615fe8e3ac5a6c4c02cc4bd4f6910b533da
---
 src/nfa/ee/nfa_ee_act.cc | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/src/nfa/ee/nfa_ee_act.cc b/src/nfa/ee/nfa_ee_act.cc
index d6991a9c6..1dc3dbe4a 100644
--- a/src/nfa/ee/nfa_ee_act.cc
+++ b/src/nfa/ee/nfa_ee_act.cc
@@ -353,24 +353,12 @@ static void nfa_ee_add_tech_route_to_ecb(tNFA_EE_ECB* p_cb, uint8_t* pp,
       power_cfg |= NCI_ROUTE_PWR_STATE_BATT_OFF;
     if ((power_cfg & NCI_ROUTE_PWR_STATE_ON) &&
         (NFC_GetNCIVersion() >= NCI_VERSION_2_0)) {
-#if (NXP_EXTNS == TRUE)
-      if ((p_cb->tech_screen_lock & nfa_ee_tech_mask_list[xx]) &&
-          (nfcFL.chipType == pn7160))
-        power_cfg |= NCI_ROUTE_PWR_STATE_SCREEN_ON_LOCK();
-      if ((p_cb->tech_screen_off & nfa_ee_tech_mask_list[xx]) &&
-          (nfcFL.chipType == pn7160))
-        power_cfg |= NCI_ROUTE_PWR_STATE_SCREEN_OFF_UNLOCK();
-      if ((p_cb->tech_screen_off_lock & nfa_ee_tech_mask_list[xx]) &&
-          (nfcFL.chipType == pn7160))
-        power_cfg |= NCI_ROUTE_PWR_STATE_SCREEN_OFF_LOCK();
-#else
       if (p_cb->tech_screen_lock & nfa_ee_tech_mask_list[xx])
         power_cfg |= NCI_ROUTE_PWR_STATE_SCREEN_ON_LOCK();
       if (p_cb->tech_screen_off & nfa_ee_tech_mask_list[xx])
         power_cfg |= NCI_ROUTE_PWR_STATE_SCREEN_OFF_UNLOCK();
       if (p_cb->tech_screen_off_lock & nfa_ee_tech_mask_list[xx])
         power_cfg |= NCI_ROUTE_PWR_STATE_SCREEN_OFF_LOCK();
-#endif
     }
     if (power_cfg) {
       if (mute_tech_route_option) {
@@ -438,14 +426,8 @@ static void nfa_ee_add_proto_route_to_ecb(tNFA_EE_ECB* p_cb, uint8_t* pp,
            enable HCE screen lock */
         if ((power_cfg & NCI_ROUTE_PWR_STATE_ON) &&
             (NFC_GetNCIVersion() >= NCI_VERSION_2_0)) {
-#if (NXP_EXTNS == TRUE)
-          if ((p_cb->proto_screen_lock & nfa_ee_proto_mask_list[xx]) &&
-              (nfcFL.chipType == pn7160))
-            power_cfg |= NCI_ROUTE_PWR_STATE_SCREEN_ON_LOCK();
-#else
           if (p_cb->proto_screen_lock & nfa_ee_proto_mask_list[xx])
             power_cfg |= NCI_ROUTE_PWR_STATE_SCREEN_ON_LOCK();
-#endif
           if (p_cb->proto_screen_off & nfa_ee_proto_mask_list[xx])
             power_cfg |= NCI_ROUTE_PWR_STATE_SCREEN_OFF_UNLOCK();
           if (p_cb->proto_screen_off_lock & nfa_ee_proto_mask_list[xx])
-- 
2.49.0

